### DISCUSSION TITLE
Questions on Docker installation method

### DISCUSSION THREAD

**Discussion by Zabooby** [excalith/excalith-start-page#34] [Q&A] [answered]
If I tried using the Docker method would I need Docker running every time I want to use the startpage?

What are the pros & cons of the Docker method?

Also I read in the faq, editing the source code can't be done with Docker , so how would I run my fork in Docker? 

I’ve been using the [Fluidity](https://github.com/PrettyCoffee/fluidity) startpage for a while, which uses GitHub pages, would it be possible to do the same with excalith or would it require too much work?

**Comment by excalith** [comment, marked as answer]
You can consider Docker as a service running on background. If you want to use Docker images, you should run this service on background. When you don't need docker, you can terminate the service. It relatively uses less resources than virtual machine, but as you can guess, this adds up with each Docker Image you use.

A simple comparison will give some hints about pros and cons between each methods

| Online | Fork |  Docker |
|  --- | --- | --- |
| Always uses latest version, so it can break your local configurations time to time due to new features. | Uses the version you forked until you want to update with upstream. Will not break anything automatically. | Uses the version of the image until you want to update. Will not break anything automatically. |
| You need to be online to use the page. | This is just the source, you will need to build and serve manually. | Docker will host locally using the optimized build just like you would do with a fork, |
| Resources will load slower depending on your connection, | You will have the freedom how serve the app, | Since it uses the localhost, it will be faster, |

In order to change the source, you need to [fork](https://github.com/excalith/excalith-start-page/wiki/Getting-Started#fork) the repository and make your changes on your own fork. Docker images are the optimized builds, ready to serve There are ways to connect to the container and make modifications, but this will become overkill and useless if you want to update it later. But with a fork, you can easily change the source and host it on a web server, start a local server or create a docker image manually. It is completely up to you. Regarding your question, you should [create a docker image manually](https://github.com/excalith/excalith-start-page/wiki/Getting-Started#docker-image) from your fork.

Personally I prefer using Docker images on my computers, and online version for my tablet. If you do not want to change anyhing from the source, you can use the Docker to set up everything easily, without using `fork` method. By doing this, you can edit the configuration either with importing a remote configuration, or editing configuration through the app.

It should be technically possible to host the start page on GitHub pages. Haven't tried GitHub pages yet, but I know there are lots of examples out there using GitHub workflows, building a NextJS app. Maybe I will have a look at that later to add it to Wiki. In the meantime, feel free to explore on your own fork and let me know how it goes!

**Reply by Zabooby** [reply]
Wow thanks for the detailed response! I'll try and fix the yarn error first and then I'll try Docker. 

Before I stop annoying you with my noob questions, could I just deploy my fork to Vercel directly? What issues could arise if I did?

Wait never mind I just read your reply in the other discussion lmao. 

Thanks again for all your help and patience!

**Reply by excalith** [reply]
No problem mate! GitHub discussions is meant to be informative and hopefully contain some learning materials.

Yes, you can deploy directly to Vercel. In order to do that, you need to create a Vercel account and bind your fork to it. This will immediately trigger the build function within the `package.json` which failed on your end, on the other discussion. Whenever you push something to `main` branch, Vercel will automatically build a new version. However, if you are testing stuff, I recommend you to create a different branch for it. This will trigger a development build, and the link you would normally use won't be affected by your test. Instead, it will create a different URL for your test branch. 

Beautiful, eh? When you get the hang of these technologies, they become companions instead of monsters. Hope you learn useful stuff for your own projects!
